### ISSUE TITLE
It seems that the URI rules are changed

### ISSUE THREAD

**Issue by boyang9602** [danieltyukov/studocuhack#15]
Unblurred is /html/bg1.png
Blurred is /html/pages/blurred/page2.webp

**Comment by danieltyukov** [comment]
@boyang9602 interesting observation, I am a little busy currently could you perhaps check if you can adjust it and implement it and make a PR request and I will test it and approve it? and reference you in the next release.

**Comment by boyang9602** [comment]
Thanks, I will try it

**Comment by danieltyukov** [comment]
@boyang9602 any luck?

**Comment by boyang9602** [comment]
Hi, I investigated it a bit, however I do not have a full solution. 
First, for the pictures, what I have mentioned above should work. However, for the documents that contains text, I do not know how to unblur the texts. 

I can share some findings.

The first example is https://www.studocu.com/en-ca/document/concordia-university/introduction-to-semiconductor-materials-and-devices/midterm-2/55562380?origin=course-trending-2. Where the first **_unblurred_** picture url is (I breakdown the parameters in different rows)
```
https://d3tvd1u91rr79.cloudfront.net/9797ea638f6c34a7aa900e571895c52c/html/bg1.png?
Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6XC9cL2QzdHZkMXU5MXJyNzkuY2xvdWRmcm9udC5uZXRcLzk3OTdlYTYzOGY2YzM0YTdhYTkwMGU1NzE4OTVjNTJjXC9odG1sXC8qLnBuZyIsIkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTcxMjM1MTEwOX19fV19
&Signature=DG-Sfyx0tGmrOlDaCQBr8hmJZyg6YwCJJ~pTSMGKlGELDozA0H1Upcw7kt-JVtoFED7QEDp--j5aF-fblH3h9AfcZXDe0BhpwOMQblFO1Oy3FXCZPl440hmlohKsWkhy0I8SmZ~KI3GdJfThh6xNvktYJRLgElVUHPCyjfn4AvJjxFmOeEldKqfNcKVDN9NJSB3irwOIfJTETfu6KOiZIO70P5SjU8ukaop5L9Tg-SAOfNp62tP2yUfwzibIXZJ4B5Z5eGoQFrtzi5XgVjciyq6JKz80iVaAV8vBg28JbN4oVhsQHZxKpJDKMrMd-A0hn~GlVZTbM2~vm~W9YqM23g__
&Key-Pair-Id=APKAJ535ZH3ZAIIOADHQ
```

The second **_blurred_** picture url is 

```
https://d3tvd1u91rr79.cloudfront.net/9797ea638f6c34a7aa900e571895c52c/html/pages/blurred/page2.webp?
Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6XC9cL2QzdHZkMXU5MXJyNzkuY2xvdWRmcm9udC5uZXRcLzk3OTdlYTYzOGY2YzM0YTdhYTkwMGU1NzE4OTVjNTJjXC9odG1sXC9wYWdlc1wvYmx1cnJlZFwvKiIsIkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTcxMjM1MTEwOX19fV19
&Signature=CV2LTqN2mzEG9~ZSRvqrOmnN7iqNZ7pRAkRMMotLJHsJ9YUZ2v3gV-WgX6sYAkfVVFercuvtn-zc945VCDvkU2rD6Wz2fFy4SrBlXOb8aFOfT671RgcK5pXjENp~t56GJVd7iA~bLTY7uKX0-cNDFKDhtIOqk8MZ4rq8xvPam5rtjedtcu~t5vNyw4LZGL~c8bDyYCB9S7A01Yt7scxrdAUsmKc-SKtVuCZKxIlsNALdJgij7mAA55RdjcGyBjpUWHMLq33k2IB2pyPX4F-DcR~vp7fEVo44s4hqSucHNmj~4QabS3sJPyz8kXzyUgqe20iNHcS4Sb2-GhAw4J4U~w__
&Key-Pair-Id=APKAJ535ZH3ZAIIOADHQ
```

We can see that 
1. The URI is different. The unblurred picture is `/html/bg1.png`, while the blurred is `/html/pages/blurred/page2.webp`.
2. The Policy and Signature are changed too. 
3. The Key-Pair-Id keeps the same. 

I tried to change the `bg1` to `bg2` in the first URL, and it is the unblurred second picture. 

However, it seems that this is a special case where there is no text. 

I tried another case (link is provided below) which consists of both background picture and texts. The above findings still work for the background pictures, however, we still cannot see the texts above the background picture. 

https://www.studocu.com/en-ca/document/concordia-university/optical-devices-for-high-speed-communications/elec425-assignment-1-solutions-explaining-how-the-question-from-assignment-1/19767622

Furthermore, by comparing the unblurred pictures of the two examples above, I found that Policy and Signatures for unblurred pictures are consistent for same document, but different for different document. Key-Pair-Id is the same for all the document. 

In summary, I only find the rules for the pictures, but I do not have a solution for the blurred texts. I am busy recently and have not had a chance to further investigate it.

**Comment by danieltyukov** [comment]
Thanks a lot for the amazing documantion. I will thoroughly go through it after my exams in a week. Will notify you after pushing some updates based on this research and further research.

Thanks again.

**Comment by boyang9602** [comment]
Thank you! Good luck with your exams!
